tasks/install-on-Debian.yml

```diff
@@ -56,3 +56,4 @@
     state: started
     enabled: true
     daemon_reload: true
+  when: buildkite_agent_allow_service_startup[ansible_os_family]
```

Comment:
This will prevent the creation of the service entirely.

To achieve what you want, make like 56 a conditional like

```yaml
state: "{{ buildkite_agent_allow_service_startup[ansible_os_family] | ternary('started', 'stopped') }}"
```